### PR TITLE
Feature/gasket-plugin-morgan

### PR DESCRIPTION
**Pull request by paulodiovani**
## Summary

Create a new plugin using `morgan` middleware to log all http requests.

## Changelog

initial version `1.0.0`

## Test Plan

- exported plugin object
- returned morgan middleware
- correct call of gasket logger function for each request log

### CODE REVIEW

File: packages/gasket-plugin-morgan/LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2019 GoDaddy Operating Company, LLC.
```
Comment (jpage-godaddy) [inline packages/gasket-plugin-morgan/LICENSE.md:3]: Can you update the copyright year?

File: packages/gasket-plugin-morgan/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# @gasket/plugin-morgan
+
+Adds morgan request logger to your appplication.
```
Comment (jpage-godaddy) [inline packages/gasket-plugin-morgan/README.md:3]:
```suggestion
Adds the `morgan` request logger to your application.
```

File: packages/gasket-plugin-morgan/lib/index.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const morgan = require('morgan');
```
Comment (jpage-godaddy) [inline packages/gasket-plugin-morgan/lib/index.js:1]: Could you add an `index.d.ts` as well so the new Gasket config properties are documented?

Comment (paulodiovani) [reply packages/gasket-plugin-morgan/lib/index.js:1]: Done, please take a look

File: packages/gasket-plugin-morgan/README.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+### Requirements
+
+- [@gasket/plugin-express]
+- [@gasket/plugin-log]
+
+### New apps
+
+```
+gasket create <app-name> --plugins @gasket/plugin-express --plugins @gasket/plugin-morgan
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-morgan/README.md:17]:
```suggestion
gasket create <app-name> --plugins @gasket/plugin-morgan
```

File: packages/gasket-plugin-morgan/lib/index.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+const split = require('split');
+const { name } = require('../package');
+
+// set log configuration in gasket.config.js, under `morgan` key
+// configuration options: http://expressjs.com/en/resources/middleware/morgan.html
+module.exports = {
+  name,
+  dependencies: ['@gasket/plugin-express', '@gasket/plugin-log'],
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-morgan/lib/index.js:9]: This should work for any plugin that executes the `middleware` lifecycle, such as [@gasket/plugin-fastify](https://github.com/godaddy/gasket/tree/main/packages/gasket-plugin-fastify). Whereas log is necessary dependency for it to work.

```suggestion
  dependencies: ['@gasket/plugin-log'],
```

File: packages/gasket-plugin-morgan/README.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  morgan: {
+    format: 'tiny',
+    options: {}
+  }
+}
+```
+
+## How it works
+
+This plugins hooks the [middleware] lifecycle from [@gasket/plugin-express] and use [Morgan] to
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-morgan/README.md:69]: Include a link for `@gasket/plugin-fastify`.

**Review by agerard-godaddy — COMMENTED** [review]
We are actively updating our plugins to fully support our Fastify plugin, in which we have [middie](https://github.com/fastify/middie) integrated. Let's make sure this new Morgan plugin works with it as well out of the box.

**Review by jpage-godaddy — APPROVED** [review]
Looks good to me after addressing @agerard-godaddy's feedback.

File: packages/gasket-plugin-morgan/README.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+Adds the `morgan` request logger to your application.
+
+[Morgan] is an HTTP request logger middleware for node.js.
+
+## Installation
+
+### Requirements
+
+- [@gasket/plugin-express]
```
[the remark this answers is not in the document]
Comment (paulodiovani) [reply packages/gasket-plugin-morgan/README.md:11]: Oh, I forgot, thanks.

**Pull request merged**